### ISSUE TITLE
Auto Update mechanism 

### ISSUE THREAD

**Issue by rluvaton** [Toinane/colorpicker#47]
Maybe this application can add auto-update mechanism for always be at the latest version

**Comment by Toinane** [comment]
Yep, it would be great too!

**Comment by rluvaton** [comment]
@Toinane I'm on fire! 🔥

**Comment by Toinane** [comment]
> @Toinane I'm on fire! 🔥

I see that 😄

**Comment by rluvaton** [comment]
> > @Toinane I'm on fire! 🔥
> 
> I see that 😄

Speaking of which, look at the new proposal I add #49

**Comment by Toinane** [comment]
Actually, this feature need to have a paid Apple Account (99$/year) and a License for Windows (min 100$) to make Colorpicker approved for auto-update. For Linux, I need to investigate more : 
- Snapcraft autoupdater (maybe possible, you can update from command line for now)
- AppImage updater (need more investigation)

**Comment by rluvaton** [comment]
> Actually, this feature need to have a paid Apple Account (99$/year) and a License for Windows (min 100$) to make Colorpicker approved for auto-update. For Linux, I need to investigate more :
> 
> * Snapcraft autoupdater (maybe possible, you can update from command line for now)
> * AppImage updater (need more investigation)

You can create script that:
1. check if you have updates in _GitHub release_
2. download the application to some temp folder (`new-version-temp`)
3. Create a copy of the application in other temp folder (`prev-version-temp`)
4. Delete the application (In `current-install-dir`)
5. Move the new application from `new-version-temp` folder to the default installation directory (`current-install-dir`)
> In case of an error, restore the previous version from `prev-version-temp`
6. Delete the `new-version-temp`
7. (Bonus) When the user open the app give him an opportunity to downgrade the app

**Comment by Toinane** [comment]
Yeah why not, but it takes a lot of effort on the different platforms. I'll see if I can do that for the next release! 

I hope this is possible without asking for administrator access for Windows or Mac

**Comment by rluvaton** [comment]
> Yeah why not, but it takes a lot of effort on the different platforms. I'll see if I can do that for the next release! 
> 
> 
> 
> I hope this is possible without asking for administrator access for Windows or Mac

Take a look at [Google Ohama tutorial](https://omaha-consulting.com/google-omaha-tutorial) (I think this is how Google Chrome seamless updates work
